sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -228,4 +244,18 @@ protected String visitSQLFunction(String funcName, String[] inputs) {
   protected String visitUnexpectedExpr(Expression expr) throws IllegalArgumentException {
     throw new IllegalArgumentException("Unexpected V2 expression: " + expr);
   }
+
+  protected String visitOverlay(String[] inputs) {
+    throw new UnsupportedOperationException("Function: OVERLAY does not support ");
```

Comment:
```suggestion
    throw new UnsupportedOperationException(s"${this.getClass.getSimpleName} does not support function: OVERLAY");
```

Reply:
why do we fail here?

Reply:
@chenzhx The default just used to display. It should return `OVERLAY` syntax.